src/server.rs

```diff
@@ -687,19 +688,17 @@ impl<C : CommandCreatorSync + 'static> ClientConnection<C> {
     /// If a full request was read, return `Ok(Some(ClientRequest))`.
     /// If data was read but not a full request, return `Ok(None)`.
     fn read(&mut self, event_loop : &mut EventLoop<SccacheServer<C>>) -> io::Result<Option<ClientRequest>> {
-        //FIXME: use something from bytes:
-        // http://carllerche.github.io/bytes/bytes/index.html
-        let mut buf : [u8; 2048] = [0; 2048];
+        let mut buf = ByteBuf::mut_with_capacity(2048);
         loop {
-            match self.sock.try_read(&mut buf) {
+            match self.sock.try_read_buf(&mut buf) {
                 Ok(None) => {
                     // Read all available data.
                     trace!("try_read returned Ok(None)");
                     break;
                 },
                 Ok(Some(n)) => {
                     trace!("try_read read {} bytes", n);
-                    self.recv_buf.extend_from_slice(&buf[..n])
+                    self.recv_buf.extend_from_slice(buf.bytes())
```

Comment:
I think what I was hoping to do here was to make `self.recv_buf` and `self.send_queue` into `MutByteBuf`, to make the logic around appending bytes and draining bytes cleaner. Does that make sense?